hazelcast/src/main/java/com/hazelcast/spi/Operation.java

```diff
@@ -309,11 +309,29 @@ public final void setWaitTimeout(long timeout) {
         setFlag(timeout != -1, BITMASK_WAIT_TIMEOUT_SET);
     }
 
+    /**
+     * @deprecated Use & override {@link #onInvocationException(Throwable)} instead.
+     */
+    @Deprecated
     public ExceptionAction onException(Throwable throwable) {
         return (throwable instanceof RetryableException)
                 ? ExceptionAction.RETRY_INVOCATION : ExceptionAction.THROW_EXCEPTION;
     }
 
+    /**
+     * Called when an <tt>Exception</tt>/<tt>Error</tt> is thrown
+     * during an invocation. Invocation process will continue, retry
+     * or fail according to returned <tt>ExceptionAction</tt> result.
+     * <p/>
+     * This method is called on caller side of the invocation.
+     *
+     * @param throwable <tt>Exception</tt>/<tt>Error</tt> thrown during invocation
+     * @return <tt>ExceptionAction</tt>
+     */
+    public ExceptionAction onInvocationException(Throwable throwable) {
```

Comment:
I didn't rename it to `onInvocationFailure`, because it doesn't have to be an invocation failure. Invocation can retry or fail. It's up to operation. And it's also inlined with returned value class: `ExceptionAction`. 

On the other hand, `onFailure` is actually an operation failure.